tests/bluetooth/tester/src/btp_bap_broadcast.c

```diff
@@ -348,14 +344,14 @@ uint8_t btp_bap_broadcast_source_setup(const void *cmd, uint16_t cmd_len,
 	/* Setup extended advertising data */
 	net_buf_simple_add_le16(&ad_buf, BT_UUID_BROADCAST_AUDIO_VAL);
 	net_buf_simple_add_le24(&ad_buf, source->broadcast_id);
-	base_ad.type = BT_DATA_SVC_DATA16;
-	base_ad.data_len = ad_buf.len;
-	base_ad.data = ad_buf.data;
-	sd.type = BT_DATA_NAME_COMPLETE;
-	sd.data_len = strlen(dev_name);
-	sd.data = (const uint8_t *)dev_name;
-	err = tester_gap_create_adv_instance(param, BTP_GAP_ADDR_TYPE_IDENTITY, &base_ad, 1,
-					     &sd, 1, &gap_settings);
+	base_ad[0].type = BT_DATA_SVC_DATA16;
+	base_ad[0].data_len = ad_buf.len;
+	base_ad[0].data = ad_buf.data;
+	base_ad[1].type = BT_DATA_NAME_COMPLETE;
+	base_ad[1].data_len = sizeof(CONFIG_BT_DEVICE_NAME) - 1;
```

Comment:
Still unsure why `sizeof` and `-1` is being used instead of `strlen` :)

Reply:
> Still unsure why `sizeof` and `-1` is being used instead of `strlen` :)

I'd assume that's because the former is build-time evaluated whereas the latter is a runtime function call. That said, it could be that compilers are able to optimize this kind of stuff for string literals.

Reply:
Fair argument if our documentation suggests it :)